libarchive/archive_read_support_format_zip.c

```diff
@@ -1766,7 +1766,13 @@ zip_read_data_zipx_lzma_alone(struct archive_read *a, const void **buff,
 	 * more than that forces the decompressor to combine reads by copying
 	 * data.
 	 */
-	compressed_buf = __archive_read_ahead(a, 1, &bytes_avail);
+	// try not to use this optimization because it may return less bytes than actually available
+	// that leads to loss of eos marker in lzma stream
+	// request exact bytes count as minimun at first
+	compressed_buf = __archive_read_ahead(a, zip->entry_bytes_remaining, &bytes_avail);
```

Comment:
This will cause `__archive_read_ahead` to try to allocate a buffer big enough to fit the entire entry in memory, even if the entry is several gigabytes in size.